pkg/api/vmtemplate/template_store.go

```diff
@@ -4,24 +4,30 @@ import (
 	"github.com/rancher/apiserver/pkg/apierror"
 	"github.com/rancher/apiserver/pkg/types"
 	ctlvmv1alpha1 "github.com/rancher/harvester/pkg/generated/controllers/harvester.cattle.io/v1alpha1"
-	"github.com/rancher/harvester/pkg/ref"
+	datautil "github.com/rancher/wrangler/pkg/data"
+	"github.com/rancher/wrangler/pkg/data/convert"
 	"github.com/rancher/wrangler/pkg/schemas/validation"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
 )
 
 type templateStore struct {
 	types.Store
-	templateVersionCache ctlvmv1alpha1.VirtualMachineTemplateVersionCache
+	templates ctlvmv1alpha1.VirtualMachineTemplateClient
 }
 
 func (s *templateStore) Update(request *types.APIRequest, schema *types.APISchema, data types.APIObject, id string) (types.APIObject, error) {
-	defaultVersionID := data.Data().String("spec", "defaultVersionId")
-	if defaultVersionID != "" {
-		ns, name := ref.Parse(defaultVersionID)
-		defaultVersion, err := s.templateVersionCache.Get(ns, name)
-		if err != nil {
-			return types.APIObject{}, apierror.NewAPIError(validation.ServerError, err.Error())
-		}
-		data.Data().SetNested(defaultVersion.Status.Version, "status", "defaultVersion")
+	tp, err := s.templates.Get(request.Namespace, request.Name, metav1.GetOptions{})
+	if err != nil {
+		return types.APIObject{}, apierror.NewAPIError(validation.ServerError, err.Error())
 	}
+
+	currentTp, err := convert.EncodeToMap(tp)
+	if err != nil {
+		return types.APIObject{}, apierror.NewAPIError(validation.ServerError, err.Error())
+	}
+
+	//not allow UI update the status
+	data.Data().SetNested(currentTp["statue"], "status")
+	data.Data().SetNested(datautil.GetValueN(currentTp, "metadata", "resourceVersion"), "metadata", "resourceVersion")
```

Comment:
Can we use [Status subresource](https://kubernetes.io/docs/tasks/extend-kubernetes/custom-resources/custom-resource-definitions/#status-subresource) for this purpose.